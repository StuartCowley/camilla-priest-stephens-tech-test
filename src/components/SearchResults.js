import React from "react";
import "../styles/search-results.css";

const SearchResults = ({ results }) => {
  if (!results) {
    return <p></p>
  } else if (!results.length) {
    return <p className="results-message">No images found</p>
  } else {
    return(
      <div className="search-results">
        {results.map((result) => (
          <img 
            key={result}
            className="search-results__image"
            data-testid="search-results__image"
            src={result}
            alt="space"
          />
        ))}  
      </div> 
    );
  }
};

export default SearchResults;