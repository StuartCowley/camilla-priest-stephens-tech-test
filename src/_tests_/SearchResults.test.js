import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  const validProps = {
    setSearchResults: jest.fn()
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchResults
        setSearchResults={validProps.setSearchResults} 
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct text", () => {
    render(
      <SearchResults
        setSearchResults={validProps.setSearchResults} 
      />
    );
    expect(screen.getByText("Search Results")).toBeInTheDocument();
  });
});

