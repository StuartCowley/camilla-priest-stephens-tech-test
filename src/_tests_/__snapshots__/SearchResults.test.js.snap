// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results"
  >
    <img
      alt="space"
      class="search-results__image"
      data-testid="search-results__image"
      src="https://unsplash.com/photos/uIf6H1or1nE"
    />
    <img
      alt="space"
      class="search-results__image"
      data-testid="search-results__image"
      src="https://unsplash.com/photos/wCujVcf0JDw"
    />
  </div>
</DocumentFragment>
`;
